lib/srv/db/postgres/schema.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v4"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	dbobjectv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1"
+	"github.com/gravitational/teleport/api/types/databaseobject"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/srv/db/common/permissions"
+)
+
+// schemaInfo represents information about all schemas in a database.
+type schemaInfo map[string]schema
+
+// schema represents single schema.
+type schema struct {
+	tables []string
+}
+
+// schemaInfoQuery is a query to return the schema info.
+const schemaInfoQuery = "SELECT schemaname, tablename FROM pg_tables"
```

Comment:
In my mind, scope filters should be applied before `ApplyDatabaseObjectImportRules`, ideally in this query. 

Does `pg_tables` return only tables in the current database? If the database isn't in the scope, then there is no need to scan it in the first place?

Reply:
> In my mind, scope filters should be applied before ApplyDatabaseObjectImportRules, ideally in this query.

This is possible, I guess. You'd have to translate our glob patterns into appropriate SQL language filters. I'm not sure if simply replacing `*` with `%` would do it or not.

> Does pg_tables return only tables in the current database? If the database isn't in the scope, then there is no need to scan it in the first place?

`pg_tables` is for current the database. We could indeed avoid the scan if we'd check the database name first.

I think both are potentially good optimisations, but I'm not sure these are worth implementing at this stage; if possible, I'd like to keep the logic simple to follow and change.

Reply:
> I think both are potentially good optimisations, but I'm not sure these are worth implementing at this stage; if possible, I'd like to keep the logic simple to follow and change.

Agreed. May leave a TODO.